pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/mercury/MercuryOrderDao.java

```diff
@@ -164,26 +163,38 @@ public HashMap<String, MercuryOrderDto> getPossibleBarcodesForParticipant(String
                         log.info("found related info about barcode " + mercuryOrderDto.getBarcode());
                         map.put(mercuryOrderDto.getBarcode(), mercuryOrderDto);
                     }
-                } catch (Exception e) {
-                    throw new RuntimeException("Error getting possible mercury orders for participant " + ddpParticipantId, e);
+                } catch (SQLException e) {
+                    throw new DsmInternalError(String.format(errorMessage, ddpParticipantId), e);
                 }
             } catch (SQLException ex) {
                 dbVals.resultException = ex;
             }
-
             return dbVals;
         });
 
         if (results.resultException != null) {
-            throw new RuntimeException("Error getting the mercury orders", results.resultException);
+            throw new DsmInternalError("Error getting the mercury orders", results.resultException);
         }
         log.info(String.format("Found %d possible barcodes for participant %s", map.size(), ddpParticipantId));
         return map;
     }
 
+    private void getKitsFromStatement(PreparedStatement stmt, HashMap<String, MercuryOrderDto> map) throws SQLException {
```

Comment:
I'm confused about this refactoring, since it seems to separate code at an odd boundary. It does not separate concerns well, but rather separates the statement execution from  statement creation, which creates an unnecessary break in the logic. The two solutions I can think of are:

1. Do not refactor. Leave the code as it was.
2. Pass in the result set rather than the statement.

Reply:
This was a sonar suggestion that I will remove